Cygwin build fails: off64_t not defined

Regression from #1106: Cygwin defines `_off64_t` but not `off64_t`

```
  CC       src/zfile.o
src/zfile.c:328:27: error: unknown type name ‘off64_t’
 zfile_seek(void *cookie_, off64_t *offset_, int whence) {
                           ^
src/zfile.c:36:31: warning: ‘zfile_seek’ used but never defined
 static cookie_seek_function_t zfile_seek;
                               ^
make: *** [Makefile:483: src/zfile.o] Error 1
```

--- a/src/zfile.c
+++ b/src/zfile.c
@@ -3,6 +3,10 @@
 #endif
 #include <sys/types.h>
 
+#ifdef __CYGWIN__
+typedef _off64_t off64_t;
+#endif
+
 #include <assert.h>
 #include <err.h>
 #include <errno.h>

